sdk/nodejs/cmd/dynamic-provider/index.ts

```diff
@@ -335,8 +338,15 @@ export function main(args: string[]): void {
         delete: deleteRPC,
         getPluginInfo: getPluginInfoRPC,
     });
-    const port: number = server.bind(`0.0.0.0:0`, grpc.ServerCredentials.createInsecure());
-
+    const port: number = await new Promise<number>((resolve, reject) => {
+        server.bindAsync(`0.0.0.0:0`, grpc.ServerCredentials.createInsecure(), (err, p) => {
+            if (err) {
+                reject(err);
+            } else {
+                resolve(p);
+            }
+        });
+    });
```

Comment:
Instead of using `async` and then all this mapping into/out of promises - perhaps just pass a callback here which does the `server.start()` and `console.log` inside the happy path (and skip the `async` entirely?

Reply:
i'm ok with async here.  and i've added the code to handle unhandled-exception/rejection.  so i think we can go with this here.  we'd still have to do the same even if this was sync+callback.

Reply:
#resolved